pkg/analysis/markerscope/initializer_test.go

```diff
@@ -0,0 +1,350 @@
+						{
+							Identifier: "custom:invalid-array",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope},
+							TypeConstraint: &markerscope.TypeConstraint{
+								AllowedSchemaTypes: []markerscope.SchemaType{markerscope.SchemaTypeArray},
+								ElementConstraint: &markerscope.TypeConstraint{
+									AllowedSchemaTypes: []markerscope.SchemaType{"invalid-type"},
+								},
+							},
+						},
+					},
+				},
+				expectedErr: `invalid type constraint: invalid element constraint: invalid schema type: "invalid-type"`,
+			}),
+
+			Entry("With Any scope (field and type)", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:flexible-marker",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope, markerscope.TypeScope},
+						},
+					},
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With override marker for built-in marker", testCase{
+				config: markerscope.MarkerScopeConfig{
+					OverrideMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "optional",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope, markerscope.TypeScope}, // Override default [FieldScope]
+						},
+					},
+				},
+				expectedErr: "",
+			}),
+
+			Entry("With override marker for non-built-in marker", testCase{
+				config: markerscope.MarkerScopeConfig{
+					OverrideMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:nonexistent",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope},
+						},
+					},
+				},
+				expectedErr: "override marker must be a built-in marker; use customMarkers for custom markers",
+			}),
+
+			Entry("With custom marker for built-in marker", testCase{
+				config: markerscope.MarkerScopeConfig{
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "optional", // Built-in marker
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope, markerscope.TypeScope},
+						},
+					},
+				},
+				expectedErr: "custom marker cannot be a built-in marker; use overrideMarkers to override built-in markers",
+			}),
+
+			Entry("With both override and custom markers", testCase{
+				config: markerscope.MarkerScopeConfig{
+					OverrideMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "optional",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope, markerscope.TypeScope},
+						},
+					},
+					CustomMarkers: []markerscope.MarkerScopeRule{
+						{
+							Identifier: "custom:marker",
+							Scopes:     []markerscope.ScopeConstraint{markerscope.FieldScope},
+						},
+					},
+				},
+				expectedErr: "",
+			}),
+		)
+	})
+
+	Context("analyzer initialization", func() {
+		It("should initialize analyzer with nil config", func() {
+			// Note: Init expects a MarkerScopeConfig, passing nil will error
+			// Use empty config instead
```

Comment:
So why not check the error?

Reply:
This is missing. The document is incorrect, so I will correct it.